### ISSUE TITLE
LUA Script error?

### ISSUE THREAD

**Issue by SFC72** [TheGameCreators/GameGuruRepo#5649]
I have been making this game for a month now with no issues, but I just got this error out of the blue today.
<img width="960" alt="Screenshot 2024-08-04 184357" src="https://github.com/user-attachments/assets/1cc418c4-8df0-4e4f-bb2d-2cfb4367e544">

**Comment by Necrym59** [comment]
I havent been able to reproduce that error can you supply information on how and when it occurred plus a screen shot of the behavior settings if possible.

**Comment by SFC72** [comment]
Steve/Necrym?
Anyways, I am so sorry I haven't gotten back to you sooner. I have been
busy with the U.S. Army out in Africa. I have attached the screenshot. I
basically went into Chat GPT and got help, sort of. The following are the
Chat GPT prompts and answers.

I have made a game with 8 mazes. It is just under 2 GB of storage. I made
it using Game Guru Max. I have a computer, a Lenovo 9i. The game keeps
closing out in the 5th level. What are my biggest problems and how can I
mitigate the issue?
ChatGPT said:

When a game consistently crashes at a specific level, several factors could
be contributing to the issue. Here are the most common problems and
potential solutions:
1. *Memory Management Issues*

   - *Problem:* The game might be consuming too much memory, causing it to
   crash.
   - *Solution:*
      - Optimize your game assets (textures, models, audio) to reduce
      memory usage.
      - Check for memory leaks in your code. Tools like Valgrind or similar
      can help detect memory leaks.
      - Ensure that unnecessary assets are not loaded into memory until
      needed.

2. *Level-Specific Bugs*

   - *Problem:* There might be bugs or errors specific to the 5th level.
   - *Solution:*
      - Review the level design and scripts for the 5th level for any
      errors or complex computations that could cause a crash.
      - Use debugging tools to step through the level and identify where
      the crash occurs.

3. *Resource Loading Issues*

   - *Problem:* The game might struggle to load resources for the 5th level.
   - *Solution:*
      - Preload assets for the level in smaller chunks to avoid overloading
      the system.
      - Check for any corrupted files or assets in the 5th level and
      replace them if necessary.

4. *Hardware Limitations*

   - *Problem:* Your hardware might not meet the game's requirements.
   - *Solution:*
      - Lower the graphics settings and resolution to reduce the load on
      your system.
      - Ensure your graphics drivers and other hardware drivers are up to
      date.

5. *Software Conflicts*

   - *Problem:* Other software running on your computer might be
   conflicting with the game.
   - *Solution:*
      - Close unnecessary applications running in the background while
      playing the game.
      - Check for any software that might interfere with Game Guru Max or
      the game itself.

6. *Game Engine Limitations*

   - *Problem:* Game Guru Max might have limitations or bugs that affect
   your game.
   - *Solution:*
      - Check the Game Guru Max forums or support for any known issues and
      patches.
      - Ensure you are using the latest version of Game Guru Max.

7. *File Size and Asset Management*

   - *Problem:* The large size of your game might be causing issues with
   asset management.
   - *Solution:*
      - Compress assets to reduce the overall game size.
      - Use level streaming techniques to load only necessary parts of the
      level at a time.

Steps to Mitigate and Troubleshoot:

   1.

   *Update and Optimize:*
   - Update Game Guru Max to the latest version.
      - Optimize game assets and scripts.
   2.

   *Debugging:*
   - Use debugging tools to pinpoint the exact cause of the crash.
      - Look for patterns in the crash, such as specific actions or events
      that trigger it.
   3.

   *Test on Different Systems:*
   - Test your game on different computers to see if the issue is specific
      to your Lenovo 9i.
   4.

   *Review and Simplify:*
   - Simplify the 5th level to see if reducing complexity helps.
      - Incrementally add elements back to identify the problematic
      component.
   5.

   *Consult Community and Support:*
   - Seek advice from the Game Guru Max community or support team.
      - Share logs or crash reports with them for more targeted assistance.

By systematically addressing these potential issues, you can identify and
resolve the problem causing your game to crash at the 5th level.
4o
You said:
Give me step by step instructions on how to optimize my Game Guru Max game.
ChatGPT said:

Optimizing your Game Guru Max game involves several steps, from asset
management to script optimization. Here’s a detailed step-by-step guide to
help you through the process:
Step 1: *Update Game Guru Max*

   1. *Check for Updates:*
      - Open Game Guru Max.
      - Navigate to the update section (usually under Help or Settings).
      - Download and install any available updates.

Step 2: *Optimize Assets*

   2.

   *Reduce Texture Sizes:*
   - Locate your texture files.
      - Use an image editing tool (e.g., Photoshop, GIMP) to reduce the
      resolution of textures without compromising quality too much.
      - Save the optimized textures and replace the originals in your
      project.
   3.

   *Compress Audio Files:*
   - Convert audio files to a more compressed format like OGG if they are
      currently WAV or other large formats.
      - Use an audio editing tool (e.g., Audacity) to reduce the bit rate
      of audio files if necessary.
   4.

   *Optimize Models:*
   - Use a 3D modeling tool (e.g., Blender) to reduce the polygon count of
      models where possible.
      - Ensure models are using efficient UV maps and textures.

Step 3: *Streamline Levels*

   5.

   *Optimize Level Design:*
   - Break large levels into smaller sections to reduce the load.
      - Use level streaming or loading zones to load only necessary parts
      of the level at a time.
   6.

   *Reduce Draw Calls:*
   - Combine static objects into a single mesh where possible to reduce the
      number of draw calls.
      - Use instances of the same object instead of multiple unique objects
      to save memory.

Step 4: *Manage Scripts and Logic*

   7.

   *Optimize Scripts:*
   - Review your scripts for any unnecessary calculations or loops.
      - Use functions and modules to keep your scripts organized and
      efficient.
      - Profile your scripts to identify and fix performance bottlenecks.
   8.

   *Use LOD (Level of Detail):*
   - Implement LOD for models so that lower-detail versions are used when
      objects are far away.
      - This reduces the amount of detail the engine needs to render at any
      given time.

Step 5: *Optimize Resource Loading*

   9.

   *Preload Assets:*
   - Preload assets for upcoming levels in smaller chunks to avoid sudden
      spikes in memory usage.
      - Use loading screens or background loading to manage asset loading
      smoothly.
   10.

   *Check for Corrupted Files:*
   - Ensure all game files are intact and not corrupted. Replace any
      problematic files.

Step 6: *Test and Debug*

   11.

   *Run Performance Tests:*
   - Use the built-in profiling tools in Game Guru Max to measure
      performance metrics.
      - Identify areas with high CPU or GPU usage and optimize accordingly.
   12.

   *Debug Crashes:*
   - Enable debug logs and check for error messages when the game crashes.
      - Use the logs to pinpoint the exact location and cause of the crash.

Step 7: *Optimize Graphics Settings*

   13. *Adjust Graphics Settings:*
      - Lower the default graphics settings (e.g., shadows, reflections)
      for better performance.
      - Provide players with options to adjust graphics settings to suit
      their hardware capabilities.

Step 8: *Consult Community and Documentation*

   14.

   *Seek Help from Community:*
   - Join Game Guru Max forums and communities to share your issues and get
      advice from other developers.
      - Check for any known issues and suggested optimizations specific to
      Game Guru Max.
   15.

   *Review Documentation:*
   - Thoroughly read through the Game Guru Max documentation for tips and
      best practices on optimization.

Step 9: *Final Testing*

   16. *Test on Multiple Systems:*
      - Test your game on different hardware configurations to ensure broad
      compatibility and performance.
      - Make adjustments based on feedback from different systems.

Step 10: *Release and Monitor*

   17. *Release and Monitor:*
      - Release your optimized game and monitor player feedback for any
      remaining issues.
      - Be prepared to release patches to address any additional
      performance problems reported by players.

By following these steps, you should be able to significantly optimize your
Game Guru Max game, reducing crashes and improving overall performance.

At that point I then went back to Chat GPT.


In Game Guru Max, is grouping objects together the same as meshing objects
for less draw calls?
ChatGPT said:

In GameGuru Max, grouping objects and meshing objects serve different
purposes and have different implications for performance.
Grouping Objects

Grouping objects together in GameGuru Max allows you to manipulate multiple
objects as a single entity. This is particularly useful for organizing your
scene and making large-scale adjustments to collections of objects. When
you group objects:

   - You can move, rotate, and scale the entire group together.
   - The individual objects within the group maintain their properties and
   can still be edited separately if needed.

However, grouping objects does not inherently reduce the number of draw
calls. Each object in the group still requires its own draw call unless
they are combined in another way.
Meshing Objects

Meshing objects, on the other hand, refers to combining multiple objects
into a single mesh. This process can significantly reduce the number of
draw calls, as the combined mesh is treated as a single object by the
rendering engine. Fewer draw calls can improve performance, especially in
scenes with a large number of objects.

In GameGuru Max:

   - Meshing reduces the overhead associated with rendering multiple
   objects.
   - It is particularly beneficial for static objects that do not need to
   be moved or altered individually once placed.

Practical Considerations

   - *Use Grouping for Organization*: Group objects for ease of
   manipulation and organization within the editor. This can simplify your
   workflow but won't impact performance significantly.
   - *Use Meshing for Performance*: Mesh objects to reduce draw calls and
   improve rendering performance, particularly in complex scenes with many
   static objects.

Steps for Meshing Objects in GameGuru Max

   1. *Select Objects*: Choose the objects you want to combine.
   2. *Mesh Command*: Use the appropriate command or tool within GameGuru
   Max to combine the selected objects into a single mesh.
   3. *Optimize*: Ensure the new mesh is optimized for your scene, checking
   for any issues that might arise from the meshing process.

Conclusion

While grouping is useful for scene management, meshing is essential for
optimizing performance through reduced draw calls. Using both techniques
appropriately will help you create efficient and manageable game
environments in GameGuru Max.
4o
You said:
give me a step by step list of how to mesh static objects, like walls and
floors in Game Guru max, include commands and places it can be found.
ChatGPT said:

Certainly! Here is a step-by-step guide on how to mesh static objects like
walls and floors in GameGuru Max:
Step-by-Step Guide to Mesh Static Objects in GameGuru Max

   1.

   *Launch GameGuru Max*
   - Open GameGuru Max from your installed programs.
   2.

   *Load Your Project*
   - Open the project where you have the static objects (walls and floors)
      that you want to mesh.
   3.

   *Select the Objects*
   - Use the selection tool to select the static objects you wish to mesh
      together. You can do this by clicking on each object while
holding down the
      Ctrl key, or by using a selection box to encompass the objects.
   4.

   *Group the Objects (Optional)*
   - If you want to organize the objects before meshing, you can group
      them. Right-click on one of the selected objects and choose Group.
      This step is optional and mainly for organizational purposes.
   5.

   *Open the Entity Properties Panel*
   - With your objects selected, open the Entity Properties Panel. This can
      usually be found on the right side of the screen or by pressing the
      shortcut E.
   6.

   *Access the Mesh Tool*
   - In the Entity Properties Panel, find the option for Combine or Mesh.
      This might be labeled as Merge or Combine Objects, depending on the
      version of GameGuru Max you are using.
   7.

   *Combine Objects*
   - Click the Combine or Mesh button to start the process. This action
      will merge the selected objects into a single mesh.
   8.

   *Optimize the Mesh*
   - After combining the objects, ensure that the new mesh is optimized.
      You can do this by checking for any overlapping faces or unnecessary
      vertices and simplifying the mesh if needed.
   9.

   *Save the Mesh*
   - Once you are satisfied with the mesh, save it. You can do this by
      right-clicking the new mesh object and selecting Save As. Choose an
      appropriate name and location for the mesh file.
   10.

   *Test the Mesh*
   - Place the newly created mesh in your scene and test it to ensure it
      works as expected. Check for any issues with rendering or performance.

Detailed Commands and Locations:

   - *Selection Tool*: Click the arrow icon in the toolbar or press Ctrl +
   click to select multiple objects.
   - *Entity Properties Panel*: This is located on the right side of the
   screen, accessible by selecting an entity and pressing E.
   - *Combine/Mesh Button*: In the Entity Properties Panel, look for Combine,
   Mesh, Merge, or Combine Objects. This option consolidates the selected
   objects into a single mesh.
   - *Save As*: Right-click the newly created mesh object, then choose Save
   As to store it in your project directory.

Additional Tips:

   - Ensure all objects you want to mesh are static and not intended for
   dynamic interactions.
   - Regularly save your project before and after making significant
   changes to avoid data loss.
   - Test the performance improvements after meshing to ensure it meets
   your expectations.

By following these steps, you should be able to efficiently mesh static
objects like walls and floors in GameGuru Max, leading to optimized
performance in your game scenes.
4o
You said:
Show me where to find the character entity properties.
ChatGPT said:

Finding the character entity properties in GameGuru Max is a
straightforward process. Here's how you can access them:
Accessing Character Entity Properties in GameGuru Max

   1.

   *Open GameGuru Max*
   - Launch the GameGuru Max application from your installed programs.
   2.

   *Load Your Project*
   - Open the project where your character entity is located.
   3.

   *Select the Character Entity*
   - Click on the character entity in your scene. This will highlight the
      character and show its properties.
   4.

   *Open the Properties Panel*
   - The Properties Panel is usually located on the right side of the
      screen. If it's not visible, you can typically open it by pressing E
      or by clicking on the properties icon in the toolbar.
   5.

   *View Character Entity Properties*
   - With the character entity selected, the Properties Panel will display
      all relevant properties for the character. This includes settings like
      health, behavior, animations, and other attributes.

Detailed Steps with Locations:

   1.

   *Launch GameGuru Max*
   - Find and launch GameGuru Max from your desktop or start menu.
   2.

   *Load Your Project*
   - Go to File > Open Project, and select your project file.
   3.

   *Select the Character Entity*
   - Click on the character in your scene view. If you have multiple
      entities, you can use the Entity Picker to select the character.
The Entity
      Picker is often found in the toolbar at the top of the screen.
   4.

   *Open the Properties Panel*
   - The Properties Panel should automatically update to show the
      properties of the selected entity. If it doesn't, you can
manually open it
      by:
         - Clicking on the Properties icon in the toolbar (usually looks
         like a gear or a list icon).
         - Or, right-clicking on the character entity and selecting
         Properties from the context menu.
      5.

   *Explore the Properties*
   - In the Properties Panel, you will see various tabs or sections, such
      as:
         - *General*: Basic properties like name, type, and tags.
         - *Behavior*: AI behavior settings, including aggression, patrol
         paths, and responses.
         - *Health and Damage*: Settings for health, damage resistance, and
         death behavior.
         - *Animations*: Options for idle, walking, attacking, and other
         animations.

By following these steps, you should be able to locate and modify the
properties of character entities in GameGuru Max. If you encounter any
difficulties, refer to the GameGuru Max documentation or community forums
for additional support.

Then....
What percentage difference does it make in less work on the processor when
you reduce the draw calls on the Game Guru Max video game that you are
making?
ChatGPT said:

Reducing draw calls in a game engine like Game Guru Max can significantly
impact the performance of the game, especially concerning the workload on
the processor (CPU) and graphics processor (GPU). Draw calls refer to the
number of times the CPU tells the GPU to draw something on the screen.
Here's a detailed explanation of how reducing draw calls can impact
performance:
Understanding Draw Calls

   1.

   *Draw Calls Defined*: A draw call is a command from the CPU to the GPU
   to render an object or a batch of objects. Each draw call has overhead
   because the CPU must prepare data, and the GPU must process this data.
   2.

   *Impact on CPU*: High draw call counts can bottleneck the CPU because it
   has to prepare and send more data to the GPU. This preparation involves
   state changes, binding resources, and other tasks that consume CPU time.
   3.

   *Impact on GPU*: While the GPU handles rendering, excessive draw calls
   can lead to inefficiencies due to the need to frequently switch contexts
   and manage multiple small batches instead of fewer large ones.

Benefits of Reducing Draw Calls

   1.

   *CPU Performance*: Reducing draw calls lowers the CPU workload, allowing
   it to process other tasks more efficiently. This can lead to smoother
   gameplay and reduced latency.
   2.

   *GPU Performance*: Fewer draw calls mean the GPU can focus on rendering
   larger batches of objects, improving rendering efficiency and potentially
   increasing frame rates.
   3.

   *Overall System Performance*: By reducing the load on both the CPU and
   GPU, the overall system can achieve better performance, including higher
   frame rates and lower power consumption.

Quantifying the Performance Gain

The percentage difference in workload reduction by decreasing draw calls
can vary based on several factors:

   1.

   *Initial Draw Call Count*: The higher the initial number of draw calls,
   the more significant the impact of reduction.
   2.

   *Hardware Specifications*: Different CPUs and GPUs will handle draw
   calls differently. High-end hardware may see less percentage gain compared
   to lower-end hardware.
   3.

   *Game Complexity*: The complexity and optimization of the game’s assets
   and scene management also play a role.

General Performance Gains

While it's challenging to provide an exact percentage without specific
data, some general observations from game development practices include:

   1.

   *Minor Reductions (10-20% fewer draw calls)*: Small reductions in draw
   calls can lead to noticeable but modest improvements in performance, often
   around 5-10% better CPU utilization and slight frame rate increases.
   2.

   *Moderate Reductions (20-50% fewer draw calls)*: More substantial
   reductions can significantly impact performance, potentially improving CPU
   workload by 15-30% and noticeably increasing frame rates.
   3.

   *Major Reductions (50%+ fewer draw calls)*: Drastically reducing draw
   calls can lead to considerable performance boosts, with CPU workload
   reductions of 30% or more and substantial frame rate improvements,
   especially on mid to low-end hardware.

Practical Example

Suppose a game initially has 1000 draw calls per frame. Reducing this by
50% to 500 draw calls could improve CPU performance by up to 30%, depending
on the specific game and hardware. This improvement could translate into
smoother gameplay, higher frame rates, and reduced stuttering.
Conclusion

Reducing draw calls is a crucial optimization technique in game
development. While the exact percentage difference in processor workload
will vary, significant reductions in draw calls can lead to substantial
performance gains, particularly in CPU-bound scenarios. For a detailed
analysis of your specific game, profiling tools and performance metrics
should be used to measure the impact of draw call reductions accurately.
4o
You said:
List some profiling tools that can measure draw calls for my Game Guru Max
game.
ChatGPT said:

Profiling tools are essential for understanding and optimizing the
performance of your game. They can help you measure draw calls, identify
bottlenecks, and optimize both CPU and GPU usage. Here are some profiling
tools that you can use with Game Guru Max or similar game engines:
1. *NVIDIA Nsight Graphics*

   - *Description*: A powerful graphics debugger and profiler for
   developers. It offers detailed insights into your GPU workloads, including
   draw calls.
   - *Features*: Real-time frame analysis, GPU performance metrics, draw
   call statistics, and debugging tools.
   - *Website*: NVIDIA Nsight Graphics
   <https://developer.nvidia.com/nsight-graphics>

2. *AMD Radeon GPU Profiler (RGP)*

   - *Description*: A low-level hardware-based GPU profiler for AMD GPUs.
   It provides detailed information about GPU workloads.
   - *Features*: Frame capture, draw call analysis, pipeline state
   visualization, and performance counters.
   - *Website*: AMD Radeon GPU Profiler

3. *Intel GPA (Graphics Performance Analyzers)*

   - *Description*: A suite of tools for analyzing and optimizing games and
   other graphics-intensive applications.
   - *Features*: Frame capture, real-time performance metrics, draw call
   analysis, and detailed GPU usage statistics.
   - *Website*: Intel GPA

4. *RenderDoc*

   - *Description*: A standalone graphics debugging tool. It's open-source
   and widely used for capturing and analyzing frame data.
   - *Features*: Frame capture, API call inspection, draw call
   visualization, and shader debugging.
   - *Website*: RenderDoc <https://renderdoc.org/>

5. *MSI Afterburner with RivaTuner Statistics Server (RTSS)*

   - *Description*: While primarily known for overclocking and monitoring,
   MSI Afterburner coupled with RTSS can provide real-time performance
   metrics, including frame rates and draw calls.
   - *Features*: Real-time performance overlay, frame rate monitoring, draw
   call statistics, and hardware utilization metrics.
   - *Website*: MSI Afterburner

6. *PIX (Performance Investigator for Xbox)*

   - *Description*: A performance tuning and debugging tool for DirectX 12
   and Xbox applications. It can be used for PC games as well.
   - *Features*: GPU and CPU performance analysis, draw call metrics, and
   detailed frame breakdowns.
   - *Website*: PIX for Windows
   <https://devblogs.microsoft.com/pix/introducing-pix-on-windows/>

7. *DirectX 12 and Vulkan Validation Layers*

   - *Description*: Both DirectX 12 and Vulkan provide built-in validation
   and debugging layers that can help profile draw calls and other performance
   metrics.
   - *Features*: Real-time debugging, draw call tracking, performance
   counters, and error reporting.
   - *Website*:
      - DirectX Developer Tools <https://devblogs.microsoft.com/directx/>
      - Vulkan Validation Layers

8. *Unity Profiler and Unreal Engine Profiler*

   - *Description*: If you are using or considering using a different
   engine like Unity or Unreal Engine for comparison, both engines have robust
   built-in profiling tools.
   - *Features*: Detailed CPU and GPU usage metrics, draw call statistics,
   real-time analysis, and optimization suggestions.
   - *Websites*:
      - Unity Profiler
      - Unreal Engine Profiler

Conclusion

Using these tools, you can gain valuable insights into the performance of
your Game Guru Max game, specifically regarding draw calls. These insights
will help you optimize your game for better performance and smoother
gameplay.

Finally this....
What does this error mean in Game guru Max and how do I fix it. Give steps
and examples.
ChatGPT said:

The error message displayed in the screenshot indicates a Lua error in
GameGuru Max. Specifically, the error states:

vbnet
Copy code
scriptbank\Community\6704278\core\puzzle\search_object.lua:113:
attempt to compare number with nil

This error occurs when the script attempts to compare a number with a value
that is nil (i.e., undefined or null). In Lua, attempting to perform
operations like comparisons with nil will cause errors, as nil is not
considered a valid number or comparable value.
Steps to Fix the Error

   1.

   *Identify the Problematic Line*:
   - The error message indicates that the issue is on line 113 of the file
      search_object.lua. Open this script in a text editor to examine the
      code.
   2.

   *Locate the Comparison*:
   - Look for a line on or around line 113 that involves a comparison, such
      as if, while, or any other condition that involves > < >= <= == or ~=.
      - For example:

      lua
      Copy code
      if someNumber > anotherValue then

      3.

   *Check the Values*:
   - Identify which variables or values are being compared. Ensure that
      both values being compared are not nil. In Lua, nil means that the
      value has not been set or is not defined.
   4.

   *Add Validity Checks*:
   - Before performing the comparison, check that the values are not nil.
      You can add a condition to ensure that both variables have valid numbers.
      - Example:

      lua
      Copy code
      if someNumber ~= nil and anotherValue ~= nil and someNumber >
anotherValue then

      - Alternatively, if one of the values can be nil and you want to set
      a default, you can use:

      lua
      Copy code
      someNumber = someNumber or 0
      anotherValue = anotherValue or 0if someNumber > anotherValue then

      5.

   *Test the Fix*:
   - Save the changes to the script and reload the level in GameGuru Max.
      Test to see if the error is resolved.

Example Fix

Suppose the problematic line in your script is:

lua
Copy code
if playerScore > highScore then

You would modify it to:

lua
Copy code
if playerScore ~= nil and highScore ~= nil and playerScore > highScore then

Or, if you want to assign default values:

lua
Copy code
playerScore = playerScore or 0
highScore = highScore or 0if playerScore > highScore then

This ensures that both playerScore and highScore are valid numbers before
comparing them.
Additional Considerations

   - *Debugging*: Add print statements to debug and check the values of the
   variables right before the comparison.

   lua
   Copy code
   print("playerScore:", playerScore, "highScore:", highScore)

   - *Logging*: If the game engine supports logging, consider logging
   errors and unexpected values for better diagnostics.

By following these steps, you should be able to resolve the "attempt to
compare number with nil" error and prevent it from occurring in the future.
If the problem persists, review the logic and flow of your script to ensure
all necessary variables are properly initialized and set before use.
4o
You said:
How do I open the Lua scripting for my game where this error occurs? please
add steps.
ChatGPT said:

To open the Lua scripting in GameGuru Max, you can follow these steps to
locate and edit the script file where the error occurs:
1. *Locate the Script File*

   - The error message indicates that the script file is located at
   scriptbank\Community\6704278\core\puzzle\search_object.lua. This path is
   relative to the GameGuru Max installation directory.

2. *Open GameGuru Max*

   - Launch GameGuru Max on your computer.

3. *Navigate to the Script Location*

   - Use the File Explorer to navigate to the GameGuru Max installation
   directory. Typically, this might be something like:

   mathematica
   Copy code
   C:\Program Files (x86)\Steam\steamapps\common\GameGuru Max

   - From the main directory, go to the scriptbank folder, then continue
   following the path:

   Copy code
   \Community\6704278\core\puzzle\


4. *Open the Script File*

   - Find the file search_object.lua in the puzzle directory.
   - Right-click on the file and select *Open With*. Choose a text editor
   such as Notepad, Notepad++, or any other code editor of your choice. For a
   more advanced experience, you might consider using an Integrated
   Development Environment (IDE) like Visual Studio Code, which provides
   syntax highlighting and other helpful features for coding.

5. *Edit the Script*

   - Once the script is open, navigate to the line number indicated by the
   error message (in this case, line 113).
   - Make the necessary adjustments to the code, as described in the
   previous response.

6. *Save the Changes*

   - After making the corrections, save the file. You can do this by
   clicking *File* > *Save* or using the shortcut Ctrl+S.

7. *Test the Changes*

   - Return to GameGuru Max and reload your game or level to test if the
   error has been resolved.

8. *Troubleshooting (if needed)*

   - If the error persists or new issues arise, revisit the script and
   check for any other potential issues. You can use debug print statements or
   other debugging methods to trace the values and flow of the program.

Additional Tips:

   - *Backup Files*: Before making changes to the script, consider creating
   a backup copy. This way, you can restore the original script if necessary.
   - *Learning Resources*: If you're unfamiliar with Lua scripting,
   consider looking up tutorials and resources to better understand the
   language and debugging processes.

By following these steps, you should be able to locate and edit the Lua
script that is causing the error in GameGuru Max.

I went and made the suspected changes but they did not work, so I changed
the script back. It's been like that with the screen shot since then at
least.
[image: Screenshot 2024-08-04 184357.png]

**Comment by Necrym59** [comment]
Chat diagnosis is pointless and generic only
We just need the specific as to where and how you used the behavior.

**Comment by SFC72** [comment]
Sorry for dumping a bunch of Chat diagnoses on you. I thought it might
help. I was wrong. I did, however, find a work around. I tried taking out
the 1st level where the LUA error occurs and putting in a later level that
I won't be using in the game. Switching the levels worked and the LUA error
didn't happen. I tried it a couple of times with a few different levels and
still no LUA error. I can safely assume that the error is somewhere in the
first level.
